assignments/this.js

```diff
@@ -13,14 +13,61 @@
 
 // code example for Window Binding
 
+function personName(name) {
+console.log(this);
+return name;
+}
+personName("Juan");
+
 // Principle 2
 
 // code example for Implicit Binding
 
+const someObj = {
+  says: "Hi",
+  sayHi: function(person) {
+    console.log(`${this.says} I am ${person}`);
+  }
+};
+someObj.sayHi("Juan");
+
 // Principle 3
 
 // code example for New Binding
 
+function someOne(sayHello) {
+  this.hi = "Hi ";
+  this.sayHello = sayHello;
+  this.speak = function () {
+  console.log(this.hi + this.sayHello);
+  console.log(this);
+  };
+}
+
+const juan = new someOne("John");
+const john = new someOne("Juan");
+
+juan.speak();
+john.speak();
+
 // Principle 4
 
-// code example for Explicit Binding
\ No newline at end of file
+// code example for Explicit Binding
+
+function someOne(sayHello) {
+  this.hi = "Hi ";
+  this.sayHello = sayHello;
+  this.speak = function () {
+  console.log(this.hi + this.sayHello);
+  console.log(this);
+  };
+}
+
+const juan1 = new someOne("John");
```

Comment:
This works. Another way to do this is to pass an object to someOne in a similar way you did for the Game.